phylo/src/substitution_models/dna_models/dna_substitution_parameters.rs

```diff
@@ -83,25 +89,19 @@ impl EvolutionaryModelParameters<DNAModelType> for DNASubstParams {
         }
     }
 
+    fn get_pi(&self) -> &FreqVector {
+        &self.pi
+    }
+
     fn set_pi(&mut self, pi: FreqVector) {
         if pi.sum() != 1.0 {
             warn!("Frequencies must sum to 1.0, not setting values");
         } else {
             self.pi = pi;
         }
     }
-}
-
-impl Display for DNASubstParams {
-    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{}", self.print_as(self.model_type))
-    }
-}
 
-impl DNASubstParams {
-    pub(crate) fn parameter_definition(
-        model_type: DNAModelType,
-    ) -> Vec<(&'static str, Vec<Parameter>)> {
+    fn parameter_definition(model_type: &DNAModelType) -> Vec<(&'static str, Vec<DNAParameter>)> {
```

Comment:
I see now... Yeah, consider returning a vector of Parameters, where parameters should have a method (or a field) that would contain that static str. It is just gonna be more readable

Reply:
The problem here is that this defines parameter sets that should be all set to equal and optimised together. 

This is an example of the output of that function:
```
vec![("alpha1", vec![Rtc]), ("alpha2", vec![Rag]), ("beta", vec![Rta, Rtg, Rca, Rcg])]
```
Otherwise, e.g., the Rta and Rtg parameters will be decoupled, changing the model to a different one in our context. I will think about how to improve this, but for now I need to prioritise the next PR.